spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/cache/CacheAutoConfigurationTests.java

```diff
@@ -81,7 +86,10 @@
  * @author Ryon Day
  */
 @ClassPathExclusions("hazelcast-client-*.jar")
-class CacheAutoConfigurationTests extends AbstractCacheAutoConfigurationTests {
```

Comment:
I can't see the connection between this change (and deleting AbstractCacheAutoConfigurationTests) and reinstating support for EhCache 3. I'd prefer to make the changes as minimal as possible and leave this as-is if we can.

Reply:
My understanding is that AbstractCacheAutoConfigurationTests was only introduced to allow the Ehcache tests to have a @ClassPathExclusions annotation on the Ehcache 2 JARs. The bug that caused that to be necessary was fixed in Ehcache 2 a while back... and regardless Ehcache 2 I assume isn't part of Spring Boot 3.x?

Reply:
Unfortunately, I can't recall the exact reason. Folding everything back into a single class is a good thing to do if we can, but it's the sort of thing that we prefer to do separately. If, as you suspect, a bug's been fixed in EhCache 2 such that the separate classes are no longer necessary we can make that change in all our currently maintained branches and not just in main.

Reply:
If I recall correctly without digging through the issue-tracker we fixed this in Ehcache 3 by shading and relocating some library version clashes that meant the two couldn't co-exist. Regardless, I'll back these changes out of this PR.